src/components/CmsZone.vue

```diff
@@ -275,40 +275,48 @@ export default class CmsZone extends Vue {
     cmsClient.trackZone({ content, zoneId: this.zoneId });
   }
 
-  private trackScrollable(contents: Content[], scrollable: Element): void {
-    if (this.zoneType === 'carousel') {
-      let timeout: number;
-      const listener = (): void => {
-        window.clearTimeout(timeout);
-        timeout = window.setTimeout((): void => {
-          const contentElm = this.$el.querySelector('.slick-current');
-          if (contentElm && this.isContentVisible(contentElm, scrollable, percentVisible)) {
-            this.trackIndex(0);
-            scrollable.removeEventListener('scroll', listener);
-          }
-        }, durationVisibleToBeTrackedMs);
+  private addScrollListener(callback: () => void | Promise<void>, delay: number = 0): () => void {
+    if (delay) {
```

Comment:
I'm pretty confused about what this `delay` does. When i run through a scenario in my head, it kinda does the same thing as a `debounce`?

Reply:
hmm, I copied this from the viewable impression tracking code. I think it tries to make sure that the content is visible for at least that amount of time before tracking it. That way if you quickly scroll past parts of the page, we won't track the contents as viewed.